### DISCUSSION TITLE
How best to update Dawn?

### DISCUSSION THREAD

**Discussion by ConduciveMammal** [Shopify/dawn#1863] [General]
I'm in the middle of developing a new theme build for a client - using Dawn as the base theme. My issue is that Dawn has been updated several times during development, with some likely essential changes.
I wonder what the best way might be to update?

I'm not using Shopify CLI, but am using Git externally from Shopify, so I thought something about utilising a Dawn `upstream`?

<img width="528" alt="CleanShot 2022-07-20 at 10 45 59@2x" src="https://user-images.githubusercontent.com/18140157/179954206-8183e34f-4c33-4a3d-93ea-455b532ac28c.png">

**Comment by chadleap** [comment]
[Staying Up to Date with Dawn Changes](https://github.com/Shopify/dawn/blob/main/README.md#staying-up-to-date-with-dawn-changes) - README

**Comment by darpansoni** [comment]
@ConduciveMammal were you able to resolve this? I am facing a similar issue and using Dawn upstream gives me ‘fatal: refusing to merge unrelated histories’ error

**Reply by tnoetzel** [reply]
I also had this issue because I started with the version from the theme store. I ended up running:

`git pull upstream main --allow-unrelated-histories --no-rebase -s recursive -X theirs`

The `-s recursive -X theirs` automatically favors all changes from the upstream, which may or may not be what you want. It worked fine for me because I only added new files rather than modifying any existing ones, but if that's not the case for you you'll probably have to manually review all changes.

**Comment by alekseykorzun** [comment]
If the theme has external modifications, I typically clone a new version into a separate directory and do a manual merge while leveraging diff. The mentioned README.me approach is only suitable for non-modified themes.

**Comment by benjyiii** [comment]
Hello guys, 

The same concern that I have we want to have Dawn Theme as our new based theme and we want to benefit from the updates is it saef to say that main branch always have the stable releases and fixes and tested thoroughly?

Thank you

**Comment by farrukh-saeed-dev** [comment]
You should add a remote upstream pointing to original Dawn repository. I am doing that and it works great.

**Comment by vitalijalbu** [comment]
Hu guys! Let's have maybe a discussion about **Dawn Showcases**! What do you think?
